massa-grpc/src/tests/stream.rs

```diff
@@ -0,0 +1,1580 @@
+        .await
+        .unwrap();
+
+    tokio::time::sleep(Duration::from_millis(50)).await;
+
+    // send block
+    block_tx.send(block_op.clone()).unwrap();
+
+    let result = tokio::time::timeout(Duration::from_secs(5), resp_stream.next())
+        .await
+        .unwrap()
+        .unwrap()
+        .unwrap();
+
+    assert!(result.signed_block.is_some());
+
+    filter_addr = massa_proto_rs::massa::api::v1::NewBlocksFilter {
+        filter: Some(
+            massa_proto_rs::massa::api::v1::new_blocks_filter::Filter::Addresses(Addresses {
+                addresses: vec!["massa".to_string()],
+            }),
+        ),
+    };
+
+    tx_request
+        .send(NewBlocksRequest {
+            filters: vec![filter_addr],
+        })
+        .await
+        .unwrap();
+
+    tokio::time::sleep(Duration::from_millis(50)).await;
+
+    let result = tokio::time::timeout(Duration::from_secs(3), resp_stream.next())
+        .await
+        .unwrap()
+        .unwrap();
+
+    assert_eq!(result.unwrap_err().message(), "invalid address: massa");
+
+    stop_handle.stop();
+}
+
+#[tokio::test]
+#[serial]
+async fn new_endorsements() {
+    let addr: SocketAddr = "[::]:4020".parse().unwrap();
+    let mut public_server = grpc_public_service(&addr);
+    let config = public_server.grpc_config.clone();
+
+    let (endorsement_tx, _endorsement_rx) = tokio::sync::broadcast::channel(10);
+
+    public_server.pool_channels.endorsement_sender = endorsement_tx.clone();
+
+    let stop_handle = public_server.serve(&config).await.unwrap();
+
+    let mut public_client = PublicServiceClient::connect(format!(
+        "grpc://localhost:{}",
+        addr.to_string().split(':').into_iter().last().unwrap()
+    ))
+    .await
+    .unwrap();
+
+    let endorsement = create_endorsement();
+
+    let (tx_request, rx) = tokio::sync::mpsc::channel(10);
+    let request_stream = tokio_stream::wrappers::ReceiverStream::new(rx);
+
+    let mut resp_stream = public_client
+        .new_endorsements(request_stream)
+        .await
+        .unwrap()
+        .into_inner();
+
+    let mut filter_ids = massa_proto_rs::massa::api::v1::NewEndorsementsFilter {
+        filter: Some(
+            massa_proto_rs::massa::api::v1::new_endorsements_filter::Filter::EndorsementIds(
+                massa_proto_rs::massa::model::v1::EndorsementIds {
+                    endorsement_ids: vec![
+                        "E1q4CBcuYo8YANEV34W4JRWVHrzcYns19VJfyAB7jT4qfitAnMC".to_string()
+                    ],
+                },
+            ),
+        ),
+    };
+
+    tx_request
+        .send(massa_proto_rs::massa::api::v1::NewEndorsementsRequest {
+            filters: vec![filter_ids],
+        })
+        .await
+        .unwrap();
+
+    tokio::time::sleep(Duration::from_millis(50)).await;
+
+    endorsement_tx.send(endorsement.clone()).unwrap();
+
+    let result = tokio::time::timeout(Duration::from_secs(2), resp_stream.next()).await;
+    assert!(result.is_err());
```

Comment:
If we split the big tests, we could separate good/bad scenario tests

Reply:
Sometimes i want to verify that it still works after receiving an error